# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/MarkupMap.h
# Commit: ceb6a73484ea
# Full Hash: ceb6a73484eada2cb70ca5255f5c9ce37ed73eae
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-03-09 09:49:21
# Regressor Bug: 1686123
# File Overlap Count: 1
# Description:
#   Bug 1686123: Fall back to ARIAGrid*Accessible for HTML table/tbody/thead/tr/td/th elements with no frame. r=morgan
#   
#   display: contents means there is no frame, which means we can't use HTMLTable*Accessible.
#   However, we can (and should) use ARIAGrid*Accessible in this case.
#   
# ==============================================================================

diff -r e682f4655ac9 -r ceb6a73484ea accessible/base/MarkupMap.h
--- a/accessible/base/MarkupMap.h	Tue Mar 09 03:10:19 2021 +0200
+++ b/accessible/base/MarkupMap.h	Tue Mar 09 00:27:44 2021 +0000
@@ -450,7 +450,7 @@
 MARKUPMAP(
     table,
     [](Element* aElement, LocalAccessible* aContext) -> LocalAccessible* {
-      if (aElement->GetPrimaryFrame() &&
+      if (!aElement->GetPrimaryFrame() ||
           aElement->GetPrimaryFrame()->AccessibleType() != eHTMLTableType) {
         return new ARIAGridAccessibleWrap(aElement, aContext->Document());
       }
@@ -488,10 +488,9 @@
         // cell accessible, because there's no underlying table layout and
         // thus native HTML table cell class doesn't work. The same is
         // true if the cell itself has CSS display:block;.
-        if (!aContext->IsHTMLTableRow() ||
-            (aElement->GetPrimaryFrame() &&
-             aElement->GetPrimaryFrame()->AccessibleType() !=
-                 eHTMLTableCellType)) {
+        if (!aContext->IsHTMLTableRow() || !aElement->GetPrimaryFrame() ||
+            aElement->GetPrimaryFrame()->AccessibleType() !=
+                eHTMLTableCellType) {
           return new ARIAGridCellAccessibleWrap(aElement, aContext->Document());
         }
         if (aElement->HasAttr(kNameSpaceID_None, nsGkAtoms::scope)) {
@@ -538,14 +537,14 @@
       if (table) {
         nsIContent* parentContent = aElement->GetParent();
         nsIFrame* parentFrame = parentContent->GetPrimaryFrame();
-        if (parentFrame && !parentFrame->IsTableWrapperFrame()) {
+        if (!parentFrame || !parentFrame->IsTableWrapperFrame()) {
           parentContent = parentContent->GetParent();
           parentFrame = parentContent->GetPrimaryFrame();
           if (table->GetContent() == parentContent &&
-              ((parentFrame && !parentFrame->IsTableWrapperFrame()) ||
-               (aElement->GetPrimaryFrame() &&
-                aElement->GetPrimaryFrame()->AccessibleType() !=
-                    eHTMLTableRowType))) {
+              ((!parentFrame || !parentFrame->IsTableWrapperFrame()) ||
+               !aElement->GetPrimaryFrame() ||
+               aElement->GetPrimaryFrame()->AccessibleType() !=
+                   eHTMLTableRowType)) {
             return new ARIARowAccessible(aElement, aContext->Document());
           }
         }